warehouse/internal/api/http.go

```diff
@@ -0,0 +1,152 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/gorilla/mux"
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	"github.com/rudderlabs/rudder-server/services/stats"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/rudderlabs/rudder-server/warehouse"
+	"github.com/rudderlabs/rudder-server/warehouse/multitenant"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+type stagingFilesRepo interface {
+	Insert(stagingFile warehouse.StagingFileT) (int64, error)
+}
+
+type WarehouseAPI struct {
+	Logger      logger.Logger
+	Stats       stats.Stats
+	Repo        stagingFilesRepo
+	Multitenant *multitenant.Manager
+}
+
+type destinationSchema struct {
+	Source      backendconfig.SourceT
+	Destination backendconfig.DestinationT
+}
+
+type stagingFileSchema struct {
+	WorkspaceID           string
+	Schema                map[string]map[string]interface{}
+	BatchDestination      destinationSchema
+	Location              string
+	FirstEventAt          time.Time
+	LastEventAt           time.Time
+	TotalEvents           int
+	UseRudderStorage      bool
+	DestinationRevisionID string
+	// cloud sources specific info
+	SourceBatchID   string
+	SourceTaskID    string
+	SourceTaskRunID string
+	SourceJobID     string
+	SourceJobRunID  string
+	TimeWindow      time.Time
+}
+
+func mapStagingFile(payload stagingFileSchema) (warehouse.StagingFileT, error) {
+	if payload.WorkspaceID == "" {
```

Comment:
@lvrach 
Any benefit to using [validators](https://pkg.go.dev/github.com/go-playground/validator/v10) for this?

Reply:
I think, eventually we need to introduce them, I find them overkill for the checks we are doing so far.